Jenkinsfile

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env groovy
+pipeline {
+    agent any
+    environment { 
+        CC = 'clang'
+    }
+    stages {
+        stage('Test e2e') {
+            steps {
+                echo 'Testing e2e ...' + env.BRANCH_NAME
+                sh 'env'
+                sh 'make teste2e'
+            }
+        }
+        stage('Build') {
+            steps {
+                echo 'Building..'
+                sh '''#!/bin/bash
+                   make lint debug release
+                '''
+            }
+        }
+        stage('Test') {
+            steps {
+                echo 'Testing..'
+                sh 'make testdebug testrelease'
+         
+            }
+        }
+        stage('Deploy') {
+            steps {
+                echo 'Deploying....'
+                sh 'make s3deploybranch' 
```

Comment:
since everything is built and tested with the branch, you could gain time if you avoided rebuilding the whole project in `/tmp`

Reply:
yes, we will optimize after everything work.